views/server-config/action-form-confirmation-view.js

```diff
@@ -100,58 +108,56 @@ export default function ActionConfirmationFormView({settings, timeframe, toggleS
-                </span>
-                <input className="micro-space" value={changedSettings.config.minDate} onChange={updateMinDate} onBlur={normalizeMinDate}/>
-                <div className="dimmed text-tiny">
-                    {(!!isValidTime && !!changedSettings.config.minDate) ?
-                        (<UtcTimestamp date={changedSettings.config.minDate}/>) : <>&nbsp;</>}
-                </div>
-            </label>
+        <div className="column column-33">
+            <div className="space">Min date of quorum update time (UTC)</div>
+            {!!isOpenMinDate && <>
+                <DateSelector value={changedSettings.config.minDate} onChange={changeMinDate} onBlur={normalizeMinDate}
+                              min={minDateUpdate} max={maxDateUpdate} className="micro-space" style={{'width': '13em'}}/>
+                <a className="icon-cancel" onClick={clearTime}/></>}
+            {!isOpenMinDate && <ManualDate onChange={toggleMinDate}/>}
         </div>
-        <div className="column column-50">
+        <div className="column column-33">
             <div className="space"/>
             <label>Expiration date of update (UTC)<br/>
-                <span className="dimmed text-tiny">
-                    (Set the date in milliseconds)
-                </span>
-                <input className="micro-space" value={changedSettings.expirationDate} onChange={changeExpirationDate}/>
-                <div className="dimmed text-tiny">
-                    {(!!isValidTime && !!changedSettings.expirationDate) ?
-                        (<UtcTimestamp date={changedSettings.expirationDate}/>) : <>&nbsp;</>}
-                </div>
+                <DateSelector value={changedSettings.expirationDate} onChange={changeExpirationDate}
+                              min={maxDateUpdate} max={maxExpirationDate} className="micro-space" style={{'width': '13em'}}/>
             </label>
         </div>
-        <div className="column column-50">
+        <div className="column">
             <div className="space"/>
             <label>Information about configuration update<br/>
                 <textarea value={changedSettings.description || ''} onChange={changeDescription} style={{marginTop: '0.55em'}}/>
             </label>
         </div>
-        <div className="column column-25 column-offset-50">
+        <div className="column column-50">
+            <label className="micro-space">
+                <input onChange={toggleAllowEarlySubmission} type="checkbox" checked={allowEarlySubmission}/>&nbsp;
+                Submit when all ready
+            </label>
+        </div>
+        <div className="column column-25">
             <Button block disabled={!isReady || inProgress} onClick={confirmUpdates}>Confirm</Button>
         </div>
         <div className="column column-25">
             <Button block outline onClick={toggleShowForm}>Cancel</Button>
         </div>
     </div>
+}
+
+export function ManualDate({onChange}) {
```

Comment:
```suggestion
export function DateInput({onChange}) {
```

Also, maybe it needs the `value` prop as well, in case if value is provided externally?